pkg/meta/meta.go

```diff
@@ -1062,7 +1062,8 @@ func GetAllNameToIDAndSpecialAttributeInfo(m *Meta, dbID int64) (map[string]int6
 			return errors.Trace(err)
 		}
 
-		res[Unescape(nameLMatch[1])] = int64(id)
+		key := Unescape(nameLMatch[1])
+		res[strings.Clone(key)] = int64(id)
```

Comment:
if `Unescape` really modified the string, we don't need to clone it anymore. How about we change line 1059 to FindSubmatch and other variables to `[]byte`, and we let `Unescape` must clone the input `[]byte` even if it's untouched?

Reply:
We clone it because after several complex operation, the  actual memory of `key` is a slice from the kvrpcpb.Response packet ... 

=> kvrpcpb.Response ...   
=> kvpair
=> key.   value  
=> ....       value []byte
=>  regexp on value to get nameLMatch
=>  nameLMatch[1]
=>  Unmarshal(nameLMatch[1]).    
=> map[key]..........   key is still referencing kvrpcpb.Response